pkg/controller/configstatus/configstatus_controller.go

```diff
@@ -0,0 +1,199 @@
+// and Start it when the Manager is Started.
+func (a *Adder) Add(mgr manager.Manager) error {
+	if !operations.IsAssigned(operations.Status) {
+		return nil
+	}
+	r := newReconciler(mgr)
+	return add(mgr, r)
+}
+
+// newReconciler returns a new reconcile.Reconciler.
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileConfigStatus{
+		// Separate reader and writer because manager's default client bypasses the cache for unstructured resources.
+		writer:       mgr.GetClient(),
+		statusClient: mgr.GetClient(),
+		reader:       mgr.GetCache(),
+
+		scheme: mgr.GetScheme(),
+		log:    log,
+	}
+}
+
+// PodStatusToConfigMapper correlates a ConfigPodStatus with its corresponding Config.
+// `selfOnly` tells the mapper to only map statuses corresponding to the current pod.
+func PodStatusToConfigMapper(selfOnly bool) handler.TypedMapFunc[*v1beta1.ConfigPodStatus] {
+	return func(_ context.Context, obj *v1beta1.ConfigPodStatus) []reconcile.Request {
+		labels := obj.GetLabels()
+		name, ok := labels[v1beta1.ConfigNameLabel]
+		if !ok {
+			log.Error(fmt.Errorf("config status resource with no mapping label: %s", obj.GetName()), "missing label while attempting to map a config status resource")
+			return nil
+		}
+		if selfOnly {
+			pod, ok := labels[v1beta1.PodLabel]
+			if !ok {
+				log.Error(fmt.Errorf("config status resource with no pod label: %s", obj.GetName()), "missing label while attempting to map a config status resource")
+			}
+			// Do not attempt to reconcile the resource when other pods have changed their status
+			if pod != util.GetPodName() {
+				return nil
+			}
+		}
+
+		return []reconcile.Request{{NamespacedName: types.NamespacedName{
+			Name:      name,
+			Namespace: obj.Namespace,
+		}}}
+	}
+}
+
+// Add creates a new config status Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	c, err := controller.New("config-status-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(
+		source.Kind(mgr.GetCache(), &v1beta1.ConfigPodStatus{}, handler.TypedEnqueueRequestsFromMapFunc(PodStatusToConfigMapper(false))))
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(source.Kind(mgr.GetCache(), &configv1alpha1.Config{}, &handler.TypedEnqueueRequestForObject[*configv1alpha1.Config]{}))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileConfigStatus{}
+
+// ReconcileConfigStatus provides the dependencies required to reconcile
+// the status of a Config resource.
+type ReconcileConfigStatus struct {
+	reader       client.Reader
+	writer       client.Writer
+	statusClient client.StatusClient
+
+	scheme *runtime.Scheme
+	log    logr.Logger
+}
+
+// +kubebuilder:rbac:groups=config.gatekeeper.sh,resources=*,verbs=get;list;watch;create;update;patch;delete
```

Comment:
This overrides existing rbac rules per https://github.com/open-policy-agent/gatekeeper/blob/master/pkg/controller/config/config_controller.go#L130-L131

should the old ones be removed?

Reply:
I think what it does is adds overlapping entries in the role rather than override?

IMO it's safest to just duplicate to avoid accidentally deleting necessary permissions due to code refactors, but not a huge deal either way for me.

Reply:
I'm ok with keeping it. Just wanted to make sure this is what we want.
> groups=config.gatekeeper.sh,resources=*,verbs=get;list;watch;create;update;patch;delete